src/shadowbox/server/manager_service.ts

```diff
@@ -146,6 +152,38 @@ export class ShadowsocksManagerService {
     }
   }
 
+  // Sets the default ports for new access keys
+  public async setPortForNewAccessKeys(req: RequestType, res: ResponseType, next: restify.Next):
+      Promise<void> {
+    const invalidPortArgument = (message: string) => new restify.BadRequestError(message);
+    try {
+      logging.debug(`setPort[ForNewAccessKeys request ${JSON.stringify(req.params)}`);
+      if (!req.params.port) {
+        return next(
+            invalidPortArgument(`Expected a port argument but found none. Request: ${req}`));
+      }
+
+      const port = req.params.port;
+      if (typeof port !== 'number') {
+        return next(invalidPortArgument(`Expected an numeric port, instead got ${port}`));
+      }
+
+      await this.accessKeys.setPortForNewAccessKeys(port);
+      this.serverConfig.data().portForNewAccessKeys = port;
+      this.serverConfig.write();
+      res.send(HttpSuccess.NO_CONTENT);
+      next();
+    } catch (error) {
+      logging.error(error);
+      if (error instanceof errors.InvalidPortNumber) {
+        return next(invalidPortArgument(error.message));
```

Comment:
I feel a little nervous about forwarding error messages like that. It's a good way to leak potentially sensitive information. It's ok to log, but it's risky to send arbitrary messages back to the client.

Maybe rename the `message` field to `reason` or `explanation`, so it's more restricted in its semantics.

Reply:
This isn't an arbitrary message, it's from InvalidPortNumber which is only ever made by us and has the stringified port number as the message.  I can't rename the message field since it's part of the Error interface.  

Are you worried that somehow this response could expose the manager server to a probing attacker?